test/e2e/operator_test.go

```diff
@@ -4253,6 +4357,20 @@ func assertIngressControllerDeleted(t *testing.T, cl client.Client, ing *operato
 	}
 }
 
+func assertIngressControllerEIPAllocationDeleted(t *testing.T, cl client.Client, ing *operatorv1.IngressController, svc *ec2.EC2) {
+	t.Helper()
+	if err := deleteIngressController(t, cl, ing, 4*time.Minute); err != nil {
+		if apierrors.IsNotFound(errors.Unwrap(err)) {
+			return
+		}
+		t.Fatalf("WARNING: cloud resources may have been leaked! failed to delete ingresscontroller %s: %v", ing.Name, err)
+	} else {
+		t.Logf("deleted ingresscontroller %s", ing.Name)
+		time.Sleep(120 * time.Second)
```

Comment:
Maybe this is a mistake, but waiting for a magic 120 seconds (I guess for the service to be cleaned up) is not a good way to do this. This function is quite redundant too, you should just use the existing function `assertIngressControllerDeleted`, and make `cleanupEIPAllocations` poll/retry for 2-3 minutes attempting to clean up all EIPs. You *could* add a function to wait for the CCM to clean up the service, but that feels like resillient than just polling to ensure EIPs are deleted.

Reply:
I tested this and it does take around 90s-120s for aws to reflect the release of EIPs.

Reply:
I will try the service retry logic you mentioned.

Reply:
So, with the current logic I am checking if the service was deleted, then poll the unassociated eips and release them.